src/chess/piece/Square.java

```diff
@@ -0,0 +1,7 @@
+package chess.piece;
+
+import chess.Position;
+
+public interface Square {
+  public boolean isValidUniqueMoveByOwnPosition(Position pos, int row, int col);
+}
```

Comment:
This interface and the way to use this doesn't make sense
I'll tell you more when face to face :+1: